### PR TITLE
Getters with default values

### PR DESCRIPTION
**Pull request by luke-jr**
These seem like they would be useful

### CODE REVIEW

File: include/univalue.h [hunk 1]
```diff
@@ -73,6 +73,22 @@ class UniValue {
     bool checkObject(const std::map<std::string,UniValue::VType>& memberTypes) const;
     const UniValue& operator[](const std::string& key) const;
     const UniValue& operator[](size_t index) const;
+    const UniValue& at(const std::string& key) const {
+        return (*this)[key];
```
Comment (MarcoFalke) [inline include/univalue.h]: In the std lib: "If pos is not within the range of the container, an exception of type std::out_of_range is thrown."

Should do the same here?

Comment (luke-jr) [reply include/univalue.h]: I'm not sure. JSON typically treats `null` and absent as identical, so returning `null` (as this currently does) *might* make sense.

**Comment by jgarzik** [comment]
Rough concept ACK - will review and test further this week, with an eye towards merging, and pushing out a new version, which is overdue

File: include/univalue.h (hunk 1 above)
Comment (promag) [reply include/univalue.h]: > JSON typically treats `null` and absent as identical

Is this in the spec? Or is it implementation specific?

Comment (luke-jr) [reply include/univalue.h]: I don't see it in any specs, but it is the current behaviour of UniValue's getters, and appears to be common among other JSON implementations as well.

Comment (promag) [reply include/univalue.h]: > Should do the same here?

I think it is fine as is.

**Pull request closed without merging**